### DISCUSSION TITLE
[Duvida]Trabalho Remoto, Freelance,   e como começar?

### DISCUSSION THREAD

**Discussion by ghost** [frontendbr/forum#1148] [Discussão]
Enfim, resolvi começar uma **Issue** para tirar uma duvida minha e que também pode ser de muitos iniciantes na área. Assim como eu.  Depois de muito tempo explorando os vários campos em desenvolvimento, decidi que no momento quero focar em Web.  
Procurei várias plataformas para o aprendizado e o que me agradou mais foi o [FreeCodeCamp](www.freecodecamp.com) .  Decidi focar meu estudo nele durante três meses.   
Para um iniciante ganhar experiência, o que, como, vocês recomendariam praticar?  E também gostaria de saber, como é o processo para arrumar um Freelance ou um Trabalho Remoto, tanto no Brasil ou no Exterior.  Qual o caminho vocês  me recomendariam seguir para atingir ambos objetivos?  Faço faculdade de ciência da computação e estou indo pro segundo semestre,  uma outra curiosidade minha que tenho, o mercado de trabalho Europeu em TI para quem não possui faculdade, mas possui passaporte é complicado? Tem a necessidade de um diploma?   A minha duvida pode parecer meio longa ou sem foco em alguns momentos(espero que não). Desculpem qualquer erro.

**Comment by klarkc** [comment]
Eu uso o upwork para freelas, mas nada é melhor do q ter uma rede de contatos para fazer jobs por fora. O upwork é legal porque o volume de trabalho é enorme, e tem de todo o tipo, você pode tentar se especializar em um ramo, e é certo que vai ter trampo. O ponto contra é o valor (eles tiram uma boa fatia do teu lucro) e a alta concorrência (massacrante, sempre vai ter um indiano cobrando mais barato para fazer o mesmo job).

**Comment by ghost** [comment]
Como eu posso começar ? Algo que eu percebi que perco muito tempo fazendo é criar um cronograma de estudos com o foco no que quero trabalhar e junto com a faculdade. Como me organizar de uma forma produtiva e criar um cronograma eficaz também? Para começar bem e produtivo.

**Comment by andersoonweb** [comment]
@brunojokubauskas de todas tuas perguntas só posso responder uma apenas. Espero que te ajude!
Em relação a como praticar vou te passar o caminho que eu também vou seguir, estou afastado da área não por opção. Acontecimentos da vida.

A pratica é muito importante, colocar a mão na massa vai te fazer aprender **MUITO**, porém, lembre-se de saber teorias, ela vai te poupar muito tempo e dor de cabeças. Ou seja, deve-se saber como executar e interessante saber o porque tbm.
HTML + CSS + Javascript: indico que comece vendo teorias e pratica, mas como praticar? Vi em um artigo há um tempo atrás e estou gostando da ideia. Você vai produzir alguns componentes - header, footer, aside, de sites aleatórias, por exemplo: abra lá o codepen e tente reproduzir esse sistema de issues, o footer do Globo Esporte, qualquer coisa. Bom, com um certo domínio você pode passar para coisas maiores, como reprodução da página inteira. Uma saída para esse processo não ficar chato, tente reproduzir redesign, você consegue achar alguns disponível no Behance e sites similares. Vale ressaltar que, o desenvolvedor gráfico vai ter toda atenção como se fosse um site real; você vai reproduzir coisas reais, se torna mais difícil enjoar ou achar chato. Dessa forma, conseguirá simular e aprender várias coisas.
Uma hora você vai montar seu _workspace_, um _start_ para começar todos os redesign, já vai ser uma puta evolução. Nesse momento, indico que você olhe pra trás e veja o quanto você aprende.

O freecodecamp é muito bom mesmo, mas dê uma atenção ao MDN também, essas documentações são bem boas para aprender por trás dos panos. E principalmente, questiona-se e busque soluções.
Leia [esse artigo ](https://medium.com/@andersoonweb/relato-de-uma-projeto-open-source-925ab40098e8)meu se for do teu interesse. Foi escrito há bastante tempo, porém, vai sentir o quanto eu evolui naquela época. Também, te indico [esse](https://medium.com/@andersoonweb/tentando-corrigir-minhas-defici%C3%AAncias-9138ee858850).

O que acho que tu deve aprender:
 - HTML;
 - CSS;
 - Javascript;
 -  Algum gerador estático - Vai te auxiliar nos projetos;
 - Pré-processadores - Sass, Stylus;
 - Task Runners - Gulp, Npm script;
 - Semântica e organização;
 - Padrões - Recomendo que olhe o Smacss;
 - Git;
 - Github;
 - Linux
Não indico por agora aprendizado de Frameworks como Bootstrap, Fondation, Pure, enfim. Eles vai te dar muita coisa pronta, infelizmente isso é ruim no começo.

Espero te ajudar de verdade cara. Muitos sucessos!

EDIT: Veja também um pouco de inglês é de extrema importância.

Alguns benefícios:
 - Acesso a bastante informações direto da fonte;
 - Falar com o mundo;
 - Consultar documentações.

**Comment by ghost** [comment]
@andersoonweb  cara,  me ajudou muito. (:

**Comment by felipefialho** [comment]
@andersoonweb Tava sumidão mano, aparece mais :)

**Comment by andersoonweb** [comment]
@LFeh, Amo essa comunidade por isso, a preocupação com o outro aqui é pré-requisito.

Cara vou aparecer sim, pode deixar! Há algum tempinho que tô fora por alguns caminhos da vida e estou trabalhando em uma outra área também, busquei fazer uma faculdade e o resultado é tempo ZERO.
Estou me programando pra retornar, já comecei a mapear os estudos e acompanhar alguns fóruns - inclusive aqui.
Pretendo também retornar e concluir o [free time](https://github.com/free-time). Tô de volta! Obrigado pela preocupação, abração!

**Comment by weltonsanches** [comment]
Cara, [este artigo](https://medium.com/trainingcenter/como-conseguir-o-seu-primeiro-freela-27bba94a7954) tem umas dicas bem legais.

**Comment by ViniUXD** [comment]
Oi @brunojokubauskas 

Cara, fica tranquilo. A confusão na sua pergunta é perfeitamente normal. Já passei por isso.

Trilhei um caminho do ZERO até atualmente ganhar $60 / h. Hoje moro na Espanha com minha esposa e trabalho 4h por dia (sim, DE VERDADE)

As minhas sugestões são:

1. Use plataformas como Upwork e freelancer.com

2. Comece com micro projetos para ganhar feebacks positivos. O seu perfil só será relevante com feedbacks. Isso é crítico. Um ou dois já valem muito.

3. Assista [este vídeo](https://youtu.be/rn4Yc1Bvp1g) (e seja receptivo à críticas):
Link no youtube: https://youtu.be/rn4Yc1Bvp1g

4. Depois do vídeo, vem pra comunidade no Facebook =) https://www.facebook.com/groups/223018108146489/

Espero ter te ajudado.

Grande abraço!

**Comment by HenriqueMorgani** [comment]
Estou querendo expandir meus conhecimentos em Python, pois vejo uma grande procura no mercado.
Encontrei um curso que me chamou atenção e gostaria de saber se alguém já fez e principalmente, se é o suficiente para iniciar.

O curso em questão é esse aqui: [https://www.udemy.com/python-gratuito/](https://www.udemy.com/python-gratuito/)
